### PR TITLE
Chore/test coverage

### PR DESCRIPTION
**Pull request by bearrito**
More test coverage.

@lemire 

I'll probably add a few more tests in on some of the less covered files. You can leave this open if you like.

### CODE REVIEW

File: bitmapcontainer_test.go [hunk 1]
```diff
@@ -485,3 +485,50 @@ func TestBitmapcontainerNextHasMany(t *testing.T) {
...
+func TestBitmapcontainerOrArrayCardinality(t *testing.T) {
+	t.Run("Empty Bitmap and Empty Array", func(t *testing.T) {
+		array := newArrayContainer()
+		bc := newBitmapContainer()
+		result := bc.orArrayCardinality(array)
+
+		assert.Equal(t, 0, result)
+	})
+
+	t.Run("Populated Bitmap with Empty Array", func(t *testing.T) {
```
Comment (bearrito) [inline bitmapcontainer_test.go:498]: @lemire  This test is failing. Seems like a bug. The following two test compare that behavior with other containers.

File: roaring64/roaringarray64_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+func TestRoaringArray64AdvanceUntil(t *testing.T) {
+	bitmap := New()
+	bitmap.AddRange(0, 128)
+	assert.Equal(t, 0, bitmap.highlowcontainer.advanceUntil(0, -1))
+	assert.Equal(t, 1, bitmap.highlowcontainer.advanceUntil(0, 0))
+	assert.Equal(t, 65, bitmap.highlowcontainer.advanceUntil(64, 0))
+}
+
+func TestRoaringArray64AdvanceUntilJavaRegression(t *testing.T) {
```
Comment (bearrito) [inline roaring64/roaringarray64_test.go:17]: Possible Regression: https://github.com/RoaringBitmap/RoaringBitmap/blob/70866e270627d29de873d4ab0d9a7bd93c16062e/RoaringBitmap/src/test/java/org/roaringbitmap/buffer/TestUtil.java#L80

File: roaringarray_test.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+func TestRoaringArrayAdvanceUntil(t *testing.T) {
+	bitmap := New()
+	bitmap.AddRange(0, 128)
+	assert.Equal(t, 0, bitmap.highlowcontainer.advanceUntil(0, -1))
+	assert.Equal(t, 1, bitmap.highlowcontainer.advanceUntil(0, 0))
+	assert.Equal(t, 65, bitmap.highlowcontainer.advanceUntil(64, 0))
+}
+
+func TestRoaringArrayAdvanceUntilJavaRegression(t *testing.T) {
```
Comment (bearrito) [inline roaringarray_test.go:17]: Regression?

File: bitmapcontainer_test.go (hunk 1 above)
Comment (lemire) [reply bitmapcontainer_test.go:498]: Indeed. It looks like you caught a bug.

Would you try the following fix at https://github.com/RoaringBitmap/roaring/blob/5aca9677d8270ffec04f4535b092216796de2135/bitmapcontainer.go#L544

```Go
func (bc *bitmapContainer) orArrayCardinality(value2 *arrayContainer) int {
	answer := bc.getCardinality()
	c := value2.getCardinality()
	for k := 0; k < c; k++ {
		// branchless:
		v := value2.content[k]
		i := uint(v) >> 6
		bef := bc.bitmap[i]
		aft := bef | (uint64(1) << (v % 64))
		answer += int((bef - aft) >> 63)
	}
	return answer
```

File: roaring64/roaringarray64_test.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRoaringArray64AdvanceUntil(t *testing.T) {
+	bitmap := New()
+	low := 1 << 32
```
Comment (lemire) [inline roaring64/roaringarray64_test.go:11]:
```suggestion
	low := uint64(1) << 32
```

File: roaring64/roaringarray64_test.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	mid := 2 << 32
```
Comment (lemire) [inline roaring64/roaringarray64_test.go:12]:
```suggestion
	mid := uint64(2) << 32
```

File: roaring64/roaringarray64_test.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	high := 3 << 32
```
Comment (lemire) [inline roaring64/roaringarray64_test.go:13]:
```suggestion
	high := uint64(3) << 32
```

File: roaring64/roaringarray64_test.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+		cow1 bool
+		cow2 bool
+	}{
+		{"AppendCopiesAfterCoW", true, true},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			r1 := 1 << 32
+			r2 := 2 << 32
```
Comment (lemire) [inline roaring64/roaringarray64_test.go:43]:
```suggestion
			r2 := uint64(2) << 32
```

File: roaring64/roaringarray64_test.go [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+			r3 := 3 << 32
```
Comment (lemire) [inline roaring64/roaringarray64_test.go:44]:
```suggestion
			r3 := uint64(3) << 32
```

File: roaring64/roaringarray64_test.go [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+			r4 := 4 << 32
```
Comment (lemire) [inline roaring64/roaringarray64_test.go:45]:
```suggestion
			r4 := uint64(4) << 32
```

**Comment by bearrito** [comment]
@lemire Looks to be passing. I likely won't be doing any more code coverage improvements soon. Please merge if everything seems in order.

**Comment by lemire** [comment]
Merging.

**Pull request merged**